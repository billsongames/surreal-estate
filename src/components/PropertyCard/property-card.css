.property-card {
  padding: 10px;
  margin: 10px;
  display: flex;
  flex-flow: column;
  justify-content: left;
  text-align: left;
  border: 2px solid black
}