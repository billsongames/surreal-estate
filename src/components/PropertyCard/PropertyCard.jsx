import React from "react";

import "./property-card.css"

function PropertyCard(props) {
  const {title, type, bathrooms, bedrooms, price, city, email} = props

  return(
    <div className="property-card">
      <div>
        Title: {title}
      </div>
      <div>
        Type: {type}
      </div>
      <div>
        Bathrooms: {bathrooms}
      </div>
      <div>
        Bedrooms: {bedrooms}
      </div>
      <div>
        Price: {price}
      </div>
      <div>
        City: {city}
      </div>
      <div>
        <a href={`mailto:${email}`}>Email</a>
      </div>
    </div>  
  )
}

export default PropertyCard