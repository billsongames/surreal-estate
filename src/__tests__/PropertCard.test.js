import { render, screen } from "@testing-library/react";

import PropertyCard from "../components/PropertyCard/PropertyCard";


describe('Property card', () => {
  const validProps = {
    fields: {
      title: "3 bed detached",
      city: "Manchester",
      type: "Detached",
      bedrooms: 0,
      bathrooms: 1,
      price: 200000,
      email: "bigbob@email.com",
    },  
    alert: {
      message: "",
      isSuccess: false
    }
  }

  test('expect test props to be correct', () => {
    const {asFragment} = render(
      <PropertyCard
        title={validProps.fields.title}
        type={validProps.fields.type}
        bathrooms={validProps.fields.bathrooms}
        bedrooms={validProps.fields.bedrooms}
        price={validProps.fields.price}
        city={validProps.fields.city}
        email={validProps.fields.email}
      />)  

    expect(asFragment()).toMatchSnapshot()

    expect(screen.getByText(/Title: 3 bed detached/)).toBeInTheDocument()
    expect(screen.getByText(/City: Manchester/)).toBeInTheDocument()
    expect(screen.getByText(/Detached/)).toBeInTheDocument()
    expect(screen.getByText(/Bedrooms/)).toBeInTheDocument()
    expect(screen.getByText(/Bathrooms/)).toBeInTheDocument()
    expect(screen.getByText(/200000/)).toBeInTheDocument()
    expect(screen.getByText(/Email/)).toBeInTheDocument()
  })
});