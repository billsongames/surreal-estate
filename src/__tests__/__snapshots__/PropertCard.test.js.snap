// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property card expect test props to be correct 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <div>
      Title: 3 bed detached
    </div>
    <div>
      Type: Detached
    </div>
    <div>
      Bathrooms: 1
    </div>
    <div>
      Bedrooms: 0
    </div>
    <div>
      Price: 200000
    </div>
    <div>
      City: Manchester
    </div>
    <div>
      <a
        href="mailto:bigbob@email.com"
      >
        Email
      </a>
    </div>
  </div>
</DocumentFragment>
`;
